src/drops.cpp

```diff
@@ -186,93 +189,82 @@ uint64_t drops::hash_data( const string data )
 {
-   eosiosystem::system_contract::sellram_action sellram{"eosio"_n, {_self, "active"_n}};
-   sellram.send(_self, bytes);
+   eosiosystem::system_contract::sellram_action sellram{"eosio"_n, {get_self(), "active"_n}};
+   sellram.send(get_self(), bytes);
 }
 
-void drops::transfer_tokens(name to, asset amount, string memo)
+void drops::transfer_tokens(const name to, const asset quantity, const string memo)
 {
-   token::transfer_action transfer_act{"eosio.token"_n, {{_self, "active"_n}}};
-   transfer_act.send(_self, to, amount, memo);
+   token::transfer_action transfer_act{"eosio.token"_n, {{get_self(), "active"_n}}};
+   transfer_act.send(get_self(), to, quantity, memo);
 }
 
-void drops::transfer_ram(name to, asset amount, string memo) { check(false, "transfer_ram not implemented"); }
+void drops::transfer_ram(const name to, const int64_t bytes, const string memo) {
+   check(false, "transfer_ram not implemented");
+}
 
-drops::drop_row drops::modify_drop_binding(name owner, uint64_t drop_id, bool bound)
+void drops::modify_drop_binding(const name ram_payer, const name owner, const uint64_t drop_id)
 {
-   drops::drop_table drops(_self, _self.value);
+   drops::drop_table drops(get_self(), get_self().value);
 
    auto drops_itr = drops.find(drop_id);
-
-   check(drops_itr != drops.end(), "Drop " + std::to_string(drops_itr->seed) + " not found");
-   check(drops_itr->owner == owner, "Drop " + std::to_string(drops_itr->seed) + " does not belong to account.");
-   check(drops_itr->bound == bound,
-         "Drop " + std::to_string(drops_itr->seed) + " bound value is already " + std::to_string(bound) + ".");
+   check_drop_ownership(owner, drop_id);
 
    // Determine the payer with bound = owner, unbound = contract
-   name ram_payer = bound ? owner : _self;
-
-   // Copy the existing row
-   drops::drop_row drop = *drops_itr;
+   const bool bound = ram_payer == owner;
 
-   // Destroy the existing row
-   drops.erase(drops_itr);
-
-   // Recreate identical drop with new bound value and payer
-   drops.emplace(owner, [&](auto& row) {
-      row.seed    = drop.seed;
-      row.owner   = drop.owner;
-      row.bound   = bound;
-      row.created = drop.created;
+   // Modify RAM payer
+   drops.modify(drops_itr, ram_payer, [&](auto& row) {
+      check(row.bound != bound, "Drop bound was not modified");
+      row.bound = bound;
    });
-
-   return drop;
 }
 
-[[eosio::action]] drops::bind_return_value drops::bind(name owner, std::vector<uint64_t> drops_ids)
+[[eosio::action]]
+drops::bind_return_value drops::bind(const name owner, const vector<uint64_t> drops_ids)
 {
    require_auth(owner);
    check_is_enabled();
 
    check(drops_ids.size() > 0, "No drops were provided to transfer.");
 
-   drops::drop_table drops(_self, _self.value);
-
    for ( const uint64_t drop_id : drops_ids ) {
-      modify_drop_binding(owner, drop_id, true);
+      check_drop_ownership(owner, drop_id);
+      check_drop_bound(owner, drop_id, false);
```

Comment:
Yes agreed, I was looking for a way to reduce those table queries to 1x but also keep those checks consistent (since they are replicated in other methods)

Reply:
Ok, should be fixed, those checks re-use the same table query lookup